### DISCUSSION TITLE
How does the `--changed` flag work

### DISCUSSION THREAD

**Discussion by ffMathy** [vitest-dev/vitest#6734] [Q&A]
How does it determine if a test is affected by a code change? Does the code change need to be directly in the given test? Does it look at the `imports`? Does it look at the code coverage files to determine which tests run through which code?

**Comment by sheremet-va** [comment]
It parses every import and creates a graph of modules. The test will run if the module is imported at any point as long as it's in the module graph of the test file.

**Reply by ffMathy** [reply]
Ah okay, thanks. It would be awesome if it could actually look at which lines each test runs through from the coverage files, and then base itself on that (assuming you'd commit the coverage files in).

That's how (for instance) Pruner works (https://github.com/pruner/cli), or IntelliJ based IDEs.

Or alternatively if it based itself on analysis of the actual code recursively.

Is this worth a feature request?

**Reply by ffMathy** [reply]
VS Code's API also supports per-test-per-line coverage: https://github.com/microsoft/vscode/issues/123713

But of course the responsibility of this probably shouldn't live in the VS Code extension.

**Reply by ffMathy** [reply]
Made a feature request here: https://github.com/vitest-dev/vitest/issues/6735
